pkg/controllers/management/auth/crtb_handler.go

```diff
@@ -256,13 +354,15 @@ func (c *crtbLifecycle) reconcileLabels(binding *v3.ClusterRoleTemplateBinding)
 			return err
 		})
 		if retryErr != nil {
+			addCondition(binding, condition, FailedToUpdateClusterRoleBindings, binding.Name, retryErr)
 			returnErr = multierror.Append(returnErr, retryErr)
```

Comment:
This could use https://pkg.go.dev/errors#Join

Reply:
Note that I added only the management of the new status field.
I intentionally did not touch the pre-existing functionality.
A quick look at `Join` and I am not convinced that this has the same semantics.
The `go-multi-error` stuff seems to manage the set of errors in `returnErr` as a flat slice.
Using `Join` naively on the other hand looks to create a linear list of wrapped errors.
Or I would have to collect an actual slice and do the `Join` at the end.
Beyond that I am not sure who deep the callers receiving the multi-error are wedded to that interface.

I would keep this as-is.

Reply:
Actually I agree with Kevin. The issue with multi-error has to do with it's license being MPL 2.0, which SUSE considers problematic. I have a PR removing it entirely from rancher: https://github.com/rancher/rancher/pull/45344

From my work, I do not believe any callers are wedded to the multi-error interface and just treat the returned multi-error as a standard error.

Reply:
Thank you for your notes and reference. ... I see that it even shows how to replace `errors.Wrapf`. The `/rbac` code has quite a lot of that.

I will use this as reference to fix the CRTB code.

Reply:
Done with commit d721dbe69dc5b137ad49d9ff68496e828584e986